src/persist-client/src/internal/paths.rs

```diff
@@ -49,6 +51,60 @@ impl PartId {
     }
 }
 
+/// A component that provides information about the writer of a blob.
+/// For older blobs, this is a UUID for the specific writer instance;
+/// for newer blobs, this is a string representing the version at which the blob was written.
+/// In either case, it's used to help determine whether a blob may eventually
+/// be linked into state, or whether it's junk that we can clean up.
+#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Clone)]
+pub enum WriterKey {
+    Id(WriterId),
+    Version(String),
```

Comment:
If `String` feels to allocatey, I'd propose replacing this with an `arrayvec::ArrayString`.

(However, I'd prefer not to replace this with `semver::Version` or similar; currently we never need to parse these strings, only compare them, which is a nice property I'd like to preserve.)